src/extra/sprite.cpp

```diff
@@ -0,0 +1,453 @@
+        else if (key[0] == 'O')
+          o = BGFX::Enums::BlendOpToBgfx(val.payload.enumValue);
+      });
+      return std::make_tuple(s, d, o);
+    };
+
+    switch (_blend.valueType) {
+    case Table: {
+      const auto [s, d, o] = getBlends(_blend);
+      state |= BGFX_STATE_BLEND_FUNC(s, d) | BGFX_STATE_BLEND_EQUATION(o);
+    } break;
+
+    case Seq: {
+      assert(_blend.payload.seqValue.len == 2);
+      const auto [sc, dc, oc] = getBlends(_blend.payload.seqValue.elements[0]);
+      const auto [sa, da, oa] = getBlends(_blend.payload.seqValue.elements[1]);
+      state |= BGFX_STATE_BLEND_FUNC_SEPARATE(sc, dc, sa, da) |
+               BGFX_STATE_BLEND_EQUATION_SEPARATE(oc, oa);
+    } break;
+
+    default: {
+      if (sheet->premultiply) {
+        state |= BGFX_STATE_BLEND_FUNC(BGFX_STATE_BLEND_ONE,
+                                       BGFX_STATE_BLEND_INV_SRC_ALPHA) |
+                 BGFX_STATE_BLEND_EQUATION(BGFX_STATE_BLEND_EQUATION_ADD);
+      } else {
+        state |= BGFX_STATE_BLEND_FUNC(BGFX_STATE_BLEND_SRC_ALPHA,
+                                       BGFX_STATE_BLEND_INV_SRC_ALPHA) |
+                 BGFX_STATE_BLEND_EQUATION(BGFX_STATE_BLEND_EQUATION_ADD);
+      }
+    } break;
+    }
+
+    bgfx::setState(state);
+
+    auto *texture =
+        reinterpret_cast<BGFX::Texture *>(_texture.get().payload.objectValue);
+    bgfx::setTexture(0, ctx->getSampler(0), texture->handle);
+
+    bgfx::submit(currentView.id, shader->handle);
+
+    // pass through
+    return input;
+  }
+
+  // note: duplicated from GFX.Draw in bgfx.cpp
+  static inline bool LayoutSetup{false};
+
+  // note: duplicated from GFX.Draw in bgfx.cpp
+  void setup() {
+    if (!LayoutSetup) {
+      PosTexCoord0Vertex::init();
+      LayoutSetup = true;
+    }
+  }
+
+  // note: duplicated from GFX.Draw in bgfx.cpp
+  struct PosTexCoord0Vertex {
+    float m_x;
+    float m_y;
+    float m_z;
+    float m_u;
+    float m_v;
+
+    static void init() {
+      ms_layout.begin()
+          .add(bgfx::Attrib::Position, 3, bgfx::AttribType::Float)
+          .add(bgfx::Attrib::TexCoord0, 2, bgfx::AttribType::Float)
+          .end();
+    }
+
+    static inline bgfx::VertexLayout ms_layout;
+  };
+
+  // note: modified from GFX.Draw in bgfx.cpp to be a real quad
+  static void createQuad(bgfx::TransientVertexBuffer *vb, float width = 1.0f,
+                         float height = 1.0f,
+                         Enums::HAlign hAlign = Enums::HAlign::Left,
+                         Enums::VAlign vAlign = Enums::VAlign::Top) {
+    if (6 ==
+        bgfx::getAvailTransientVertexBuffer(6, PosTexCoord0Vertex::ms_layout)) {
+      bgfx::allocTransientVertexBuffer(vb, 6, PosTexCoord0Vertex::ms_layout);
+      PosTexCoord0Vertex *vertex = (PosTexCoord0Vertex *)vb->data;
+
+      const float zz = 0.0f;
+
+      // note: alignments define the anchor point
+      const float minx = ((int)Enums::HAlign::Left - (int)hAlign) * width / 2;

```

Comment:
For pixel perfect sprites this should be floor/ceiled so it doesn't get .5 fractional coordinates
Maybe it can be the default behavior but turned of with `:Snap false` or `:AlignToGrid false`, etc.

Reply:
Good point.

Reply:
Looking at it again, in the current `sprite` branch, the coordinates are normalized between 0 and 1, so can't use floor/ceiling there.

However, whatever solution we decide to use to fix the rotation issue (either branches `sprite_camera`, `sprite_camera2` or another solution), we can think of adding this "snap to grid" option.